cli/awsbatch/awsbstat.py

```diff
@@ -299,22 +300,59 @@ def run(self, job_status, expand_children, job_queue=None, job_ids=None, show_de
         else:
             fail("Error listing jobs from AWS Batch. job_ids or job_queue must be defined")
 
+        sort_keys_function = self.__sort_by_status_startedat_jobid() if not job_ids else self.__sort_by_key(job_ids)
         if details_required:
-            self.output.show()
+            self.output.show(sort_keys_function=sort_keys_function)
         else:
-            table_keys = ["jobId", "jobName", "status", "startedAt", "stoppedAt", "exitCode"]
-            # the lambda maps an entry of the output table to its position in the AWS_BATCH_JOB_STATUS list.
-            # This makes it so that the rows in the output are sorted by (status, startedAt, jobId).
             self.output.show_table(
-                keys=table_keys,
-                sort_keys_function=lambda x: (
-                    AWS_BATCH_JOB_STATUS.index(x[table_keys.index("status")]),
-                    x[table_keys.index("startedAt")],
-                    x[table_keys.index("jobId")],
-                ),
+                keys=["jobId", "jobName", "status", "startedAt", "stoppedAt", "exitCode"],
+                sort_keys_function=sort_keys_function,
             )
 
-    def __populate_output_by_job_ids(self, job_ids, details):
+    @staticmethod
+    def __sort_by_key(ordered_keys):  # noqa: D202
+        """
+        Build a function to sort the output by key.
+
+        :param ordered_keys: list containing the sorted keys.
+        :return: a function to be used as key argument of the sorted function.
+        """
+
+        def _sort_by_key(item):
+            job_id = item.id
+            try:
+                # in case the parent id was provided as input, sort children based on parent id position
+                job_position = ordered_keys.index(job_id[: AWSBstatCommand.__JOB_ID_LENGTH])
```

Comment:
We could split by space instead of using a fixed length

Reply:
Replaced with: `parent_id = re.findall(r"[\w-]+", job_id)[0]`